extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java

```diff
@@ -184,70 +201,178 @@ public void forSubResourceWebTarget(MethodCreator methodCreator, IndexView index
             ClassInfo subInterfaceClass, MethodInfo rootMethod, MethodInfo subMethod,
             AssignableResultHandle webTarget, BuildProducer<GeneratedClassBuildItem> generatedClasses) {
 
-        Map<String, QueryData> queryParamsByName = new HashMap<>();
-        collectClientQueryParamData(rootInterfaceClass, rootMethod, queryParamsByName);
-        collectClientQueryParamData(subInterfaceClass, subMethod, queryParamsByName);
+        Map<String, ParamData> queryParamsByName = new HashMap<>();
+        collectClientParamData(rootInterfaceClass, rootMethod, queryParamsByName,
+                CLIENT_QUERY_PARAM, CLIENT_QUERY_PARAMS, "ClientQueryParam");
+        collectClientParamData(subInterfaceClass, subMethod, queryParamsByName,
+                CLIENT_QUERY_PARAM, CLIENT_QUERY_PARAMS, "ClientQueryParam");
+
         for (var headerEntry : queryParamsByName.entrySet()) {
             addQueryParam(subMethod, methodCreator, headerEntry.getValue(), webTarget, generatedClasses, index);
         }
     }
 
-    private void collectClientQueryParamData(ClassInfo interfaceClass, MethodInfo method,
-            Map<String, QueryData> headerFillersByName) {
-        AnnotationInstance classLevelHeader = interfaceClass.declaredAnnotation(CLIENT_QUERY_PARAM);
-        if (classLevelHeader != null) {
-            headerFillersByName.put(classLevelHeader.value("name").asString(),
-                    new QueryData(classLevelHeader, interfaceClass));
+    @Override
+    public AssignableResultHandle handleFormParams(MethodCreator methodCreator, IndexView index, ClassInfo interfaceClass,
+            MethodInfo method, BuildProducer<GeneratedClassBuildItem> generatedClasses, AssignableResultHandle formParams,
+            boolean multipart) {
+
+        Map<String, ParamData> formParamsByName = new HashMap<>();
+        collectClientParamData(interfaceClass, method, formParamsByName,
+                CLIENT_FORM_PARAM, CLIENT_FORM_PARAMS, "ClientFormParam");
```

Comment:
Please, move "ClientFormParam" into a constant and/or provide a custom collectClientParamData method only for form params.